api/v1alpha1/virtualmachineclass_types.go

```diff
@@ -21,6 +21,31 @@ type DynamicDirectPathIODevice struct {
 	CustomLabel string `json:"customLabel,omitempty"`
 }
 
+// InstanceStorage encapsulates information about instance storage.
+type InstanceStorage struct {
+	// StorageClass describes the configuration of instance storage related attributes on the VirtualMachine instance.
+	// +optional
+	StorageClass string `json:"storageClass,omitempty"`
+
+	// Volumes describes a list of InstanceStorageVolumes that are to be configured on the VirtualMachine.
+	Volumes []InstanceStorageVolume `json:"instanceStorage"`
+}
+
+// InstanceStorageVolume contains configuration corresponding to an instance storage volume.
+type InstanceStorageVolume struct {
+	Spec InstanceStorageVolumeSpec `json:"spec"`
+}
+
+// InstanceStorageVolumeSpec describes an instance storage volume specification.
+type InstanceStorageVolumeSpec struct {
+	Resources InstanceStorageVolumeResource `json:"resources"`
+}
+
+// InstanceStorageVolumeResource describes an instance storage volume resource specification.
+type InstanceStorageVolumeResource struct {
+	Requests resource.Quantity `json:"requests"`
+}
+
```

Comment:
`InstanceStorageVolumeSpec` and `InstanceStorageVolumeResource` seems verbose and unnecessary. 

Can we omit those and just have: 
```
type InstanceStorageVolume struct {
  Requests resource.Quantity `json:"requests"`
}
```

Reply:
Hard agree. Non CRDs do not need to follow the spec pattern. Please refactor to what @janitha recommends above.

Reply:
IMO `Size` would much more descriptive field name

Reply:
In addition, please indicate with a comment the unit of size.